### ISSUE TITLE
Gran Turismo 6 1.05 - Poor performance on NVIDIA GPU

### ISSUE THREAD

**Issue by Wolver32** [RPCS3/rpcs3#15841]
### Quick summary

I recently swapped my RX 7800 XT for a RTX 4070 and I noticed that Gran Turismo 6 performance doesn't match at all what I was getting with the AMD card. I played GT6 extensively with my old RX 7800 XT and I know that this class of cards is more than capable of doing 4K (300% resolution scale from 720p) at 60 FPS. Here's what I'm getting with my new RTX 4070:
- 300% resolution scale
![Screenshot 2024-07-27 114946](https://github.com/user-attachments/assets/83de787b-f682-4a3b-a07b-1ae2bc46acfd)
GPU utilization is constantly at 100% even though power consumption is at around 90-100W (this card consumes 200W under full load), framerate is around 25-30

- 200% resolution scale
![Screenshot 2024-07-27 115243](https://github.com/user-attachments/assets/d6b2d678-a1e0-42b0-a365-cfe39fbfcc51)
GPU utilization and power remain more or less the same, but this time I'm closer to 60 FPS

- Native resolution (720p)
![Screenshot 2024-07-27 115441](https://github.com/user-attachments/assets/094d0139-991f-4d7b-ba71-22676d108557)
Now I get a locked 60 FPS, but GPU utilization still seems high considering I'm rendering at native resolution

For comparison, this is what I'm getting on Gran Turismo 5 at 4K60 on my RTX 4070:
![Screenshot 2024-07-27 115845](https://github.com/user-attachments/assets/287cd9fd-7ff4-4f45-bc4f-368cc29b1a01)
Locked 60 FPS with the expected GPU utilization / power consumption

For both games I used the suggested settings in the RPCS3 Wiki, GT5 is at version 2.17 while GT6 is at version 1.05. These screenshots were captured from the most recent RPCS3 builds on Windows at the time of writing. I tried using an older build (8-9 months old), but I was getting the exact same results. I also tested this on Linux with the proprietary NVIDIA drivers, I thought maybe there was something weird going on with my Windows install... Well nope, same results for both games.
I don't know if this issue is a RTX 40-Series exclusive, from what I remember GT6 used to play fine on a RTX 3070, but don't count me on that. I am however sure that a RX 7800 XT (which is very close in terms of performance) was absolutely able to provide a locked 60 FPS at 4K.
CPU is an i5-12600KF. I had the exact same CPU with my old 7800 XT.

### Attach a log file

Not needed

### System configuration

CPU: Intel Core i5-12600KF
Mobo: MSI PRO Z690-A DDR4
RAM: 2x16GB Corsair Vengeance DDR4 3200MT/s CL16
GPU: NVIDIA RTX 4070
GPU Driver Version (Windows): 560.70
GPU Driver Version (Linux): 555.58.02

**Comment by Fargolini** [comment]
Just so you know, this game is more CPU heavy than GPU heavy, I would imagine at this stage clock speeds and IPC make more of a difference over a better GPU. Plus there are still a boat load of bugs and crashes with the Gran Turismo games as a whole anyway. At this moment in time you're best off either overclocking, getting the best CPU in your budget if you choose to upgrade, or wait for the emulator to become more optimised, it seems like you know what settings work well with the game so unfortunately I don't know what else to do here apart from posting a log and seeing if a dev can find a specific bug :/

**Comment by kd-11** [comment]
NVIDIA refuses (or their hardware is unable) to support shader stencil export. This means MSAA is very heavy on NVIDIA if depth buffer writes are involved. There are heuristics in place to avoid writing MSAA images through image store as much as possible, but it is not always possible to avoid all cases. This means to load MSAA data from raw VRAM to a framebuffer costs 8x more on NVIDIA than the competitors and there is nothing we can do about it. Most big GPUs (which are NVIDIA's focus these days) can just brute-force it and eat the penalty but once you start upscaling to 4k (8k internal due to MSAA) you're asking too much of a 4070.
You can disable MSAA and just use res scaling instead if you are affected. At 4k and no MSAA the speckling that affects the mesh gratings is minimal enough that you should be ok.

TL;DR - Not our bug, hardware feature missing.
